src/call/github-workflow-runs-monitor.ts

```diff
@@ -70,4 +81,35 @@ export default async function githubWorkflowRunsMonitor(app: Probot, context: an
   } catch (error) {
     app.log.error(`Error indexing log data: ${error}`);
   }
+
+  let count: any;
+  if (job?.status === 'completed' && job?.conclusion === 'success') {
+    count = 0;
+  }
+  if (job?.status === 'completed' && job?.conclusion === 'failure') {
+    count = 1;
+  }
+  try {
+    const cloudWatchClient = new CloudWatchClient({ region: String(process.env.REGION) });
+    const putMetricDataCommand = new PutMetricDataCommand({
```

Comment:
Just wonder should we put these cloudwatch index in automation-app, or in metricscluster.

Reply:
This should be part of the automation app, github-workflow-runs-monitor.ts, resource Peter. We need to trigger a CloudWatch alarm immediately when a failure is detected. While most of the alarm setup is in the metrics cluster (and in the metrics repo: https://github.com/opensearch-project/opensearch-metrics/pull/82), we simply need to add a count to the alarm and index the data into the cluster. Monitoring workflows asynchronously after indexing to the metrics cluster would be too delayed. For instance, once data is added, we'd have to run a Lambda to query for failures (which is too late), or configure the alerting plugin to find the documents and alert us. However, the alerting plugin doesn’t easily capture passed metrics. Therefore, a straightforward solution would be to update the CloudWatch alarm immediately upon event detection.

Reply:
I wonder if we should write a separate utility to handle cloudwatch, just like `utility/opensearch/opensearch-client`.

Reply:
Initially I thought that peter, but for now except for src/call/github-workflow-runs-monitor.ts I dont see any other use for now, in future we can always move to utility/opensearch/opensearch-client.

Reply:
I think we should always try to make these code re-usable.
Ex: Maintainers dashboards can face similar issue on this.

Reply:
Can you please start with an issue 1st.

Reply:
Ya then we can seperate into utility based on the use case peter.